src/vmm/src/vstate/vcpu/mod.rs

```diff
@@ -241,6 +242,8 @@ impl Vcpu {
         let vcpu_thread = thread::Builder::new()
             .name(format!("fc_vcpu {}", self.kvm_vcpu.index))
             .spawn(move || {
+                mask_handled_signals().expect("Unable to install signal mask on VCPU thread.");
```

Comment:
Very nice catch. VCPU threads inherit the mask from the VMM thread, since they are created after the signalfd is registered.

The API thread is the one who inherits empty signal masks, since at the time it is launched, the `signalfd` was not yet created. And we don't want to mask the signals unless we have the signalfd mechanism in place (to not lose any signals data).

Reply:
Removed this